### PR TITLE
fix(grpc_client): warn instead of error on conflicting constraints

### PR DESCRIPTION
**Pull request by CatherineSue**
### Problem

When both `response_format` and `tool_call_constraint` are present, the gRPC clients return a hard error (`"Constrained decoding is not compatible with tool calls."`). However, the HTTP serving layers in all three backends handle this as a non-fatal conflict:

| Backend | HTTP behavior |
|---|---|
| **SGLang** | `logger.warning(...)`, response_format wins, tool constraint dropped |
| **vLLM** | Silently clears `response_format`, tool constraint wins |
| **TRT-LLM** | Silently skips tool constraint if `guided_decoding` already set by response_format |

The gRPC path should match the HTTP behavior rather than being stricter.

Additionally, TRT-LLM's gRPC client had the opposite priority order from its HTTP layer (tool constraint won instead of response_format).

### Solution

Change each gRPC client to warn and resolve the conflict matching its HTTP backend:

- **SGLang**: response_format wins, tool constraint dropped (warning logged)
- **vLLM**: tool constraint wins, response_format cleared (warning logged)
- **TRT-LLM**: response_format wins, tool constraint skipped (warning logged). Also fixes priority order to match HTTP behavior.

## Changes

- `sglang_scheduler.rs`: `Err(...)` → warn + drop tool constraint
- `vllm_engine.rs`: `Err(...)` → warn + `constraints.clear()` + push tool constraint
- `trtllm_service.rs`: flip priority so response_format is checked first, warn if both present

## Test Plan

Existing tests pass. This is an edge case (users rarely send both response_format and tools), but now the gRPC path matches the HTTP path instead of erroring.

<details>
<summary>Checklist</summary>

- [x] `cargo +nightly fmt` passes
- [x] `cargo clippy --all-targets --all-features -- -D warnings` passes
- [ ] (Optional) Documentation updated
- [ ] (Optional) Please join us on Slack [#sig-smg](https://slack.lightseek.org) to discuss, review, and merge PRs

</details>

## Summary by CodeRabbit

* **Bug Fixes**
  * No longer error when tool calls are used alongside other response-format/structured constraints. Conflicts are now resolved deterministically: in some flows the tool constraint is ignored (with a warning), in others it takes precedence (with a warning). Unknown tool-call types still produce an error.

### CODE REVIEW

**Review by slin1237 — APPROVED** [review]

File: crates/grpc_client/src/sglang_scheduler.rs [hunk 1]
```diff
@@ -512,21 +512,26 @@ impl SglangSchedulerClient {
...
+                        proto::sampling_params::Constraint::JsonSchema(constraint_value)
+                    }
+                    "ebnf" => proto::sampling_params::Constraint::EbnfGrammar(constraint_value),
+                    "regex" => proto::sampling_params::Constraint::Regex(constraint_value),
+                    _ => return Err(format!("Unknown constraint type: {constraint_type}")),
+                };
+                constraints.push(tool_constraint);
+            } else {
+                warn!("Constrained decoding is not compatible with tool calls, dropping tool constraint");
             }
```
[the remark this answers is not in the document]
Comment (CatherineSue) [reply crates/grpc_client/src/sglang_scheduler.rs:534]: In thoery, tool constraint is generated by the grpc preparation stage. It won't contain invalid contraint type.
